File: app/src/main/java/com/elvotra/clean/data/local/TypicodeDao.java
package com.elvotra.clean.data.local;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.elvotra.clean.data.local.model.CommentEntity;
import com.elvotra.clean.data.local.model.PostEntity;
import com.elvotra.clean.data.local.model.UserEntity;

import java.util.List;

@Dao
public interface TypicodeDao {

    @Query("SELECT * FROM users")
    List<UserEntity> getUsers();

    @Query("SELECT * FROM users WHERE users_id = :userId")
    UserEntity getUserById(long userId);

    @Query("SELECT * FROM posts ORDER BY user_id")
    List<PostEntity> getUsersPosts();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertPost(List<UserEntity> userEntities, List<PostEntity> postEntities, List<CommentEntity> commentEntities);

    @Query("DELETE FROM users WHERE users_id = :userId")
    void deleteUser(long userId);

    @Query("SELECT * FROM comments ORDER BY post_id")
    List<CommentEntity> getComments();
}

File: app/src/main/java/com/elvotra/clean/data/local/TypicodeDao.kt
package com.elvotra.clean.data.local

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

import com.elvotra.clean.data.local.model.CommentEntity
import com.elvotra.clean.data.local.model.PostEntity
import com.elvotra.clean.data.local.model.UserEntity

@Dao interface TypicodeDao {

    @get:Query("SELECT * FROM users")
    val users: List<UserEntity>

    @get:Query("SELECT * FROM posts")
    val posts: List<PostEntity>

    @get:Query("SELECT * FROM comments ORDER BY post_id")
    val comments: List<CommentEntity>

    @Query("SELECT * FROM posts WHERE posts_id = :postId")
    fun getPostById(postId: Int): PostEntity

    @Query("SELECT * FROM users WHERE users_id = :userId")
    fun getUserById(userId: Int): UserEntity

    @Query("SELECT * FROM comments WHERE post_id = :postId")
    fun getCommentsByPostId(postId: Int): List<CommentEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertPost(post: PostEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUser(user: UserEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertComment(comment: CommentEntity)

    @Query("DELETE FROM users")
    fun deleteUsers()

    @Query("DELETE FROM posts")
    fun deletePosts()

    @Query("DELETE FROM comments")
    fun deleteComments()
}
